plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/internal/AppYamlValidator.java

```diff
@@ -44,9 +47,13 @@
   private final IPath basePath;
   private final IObservableValue appYamlPath;
 
-  public AppYamlValidator(IPath basePath, IObservableValue appYamlPath) {
+  public AppYamlValidator(IPath basePath, Text fileField) {
```

Comment:
I think you should try to keep this validator SWT-agnostic.  More to the point, we may have the file field being computed in a different way.

(TBH, the basePath could also be an observable value, though we'd just use a `WritableValue`)

Reply:
I'll make both constructors `public`. That way, the validator is SWT-agnostic by its nature, and the SWT-constructor is just a convenience method.

I think keeping `basePath` as a constant makes understanding code easy. However, I might find some usefulness in the near future, so let me think about it.

Reply:
I'll keep `basePath` as `IPath`. Making it a databinding observable value (which implies a dynamically changing value) adds cognitive complexity whose feature is not used at all.